### ISSUE TITLE
Accelerometer calibration not performed successfully. Locked in step 4.

### ISSUE THREAD

**Issue by rodolfot** [iNavFlight/inav#4289]
I have a SP Racing F3 Deluxe and after updating it with the latest version of INAV and trying to calibrate the accelerometer, it always stays in step 4. Even if I turn the controller in other directions only step 4 is calibrated and when saved the configuration is as if it had done nothing, because the range values remain the same as the default.

I looked at the step-by-step how to calibrate the sensors (accelerometer, compass, gps) and even without success.

I also tried to install some previous versions of INAV but without success.

Could anyone give me a light if it is defective from my boards (2 SP Racing F3 with the same problem? Is it? Lack of luck ?!) or defect from the INAV version?

Thank you very much.

**Comment by teknoflyer** [comment]
I have  the spracing f3 EVO which flew great back when it had 1.2 on it. the calibration works fine but im still trying to find the settings on 2.01 that will make it as stable as it was on 1.2 . I just went in the original order for the different 6 axis and it completed fine. make sure you do a full chip erase during flashing.  other than that it should work. . I think I also have the F3 deluxe . will try it tomorrow and let you know

**Comment by rodolfot** [comment]
Hi, @teckel12!
- The funny thing is that I set the accelerometer in the Cleanflight and the drone "drawing" was stable / stopped. When I set up with INAV, the drone's drawing moves very slowly. I understand that this behavior occurs because it was not calibrated correctly.
- Soon, I should post more information and pictures about the tests I did.
- If I try to calibrate the compass the values are zeroed, ie it is not calibrating either.
- Always update the firmware of the controller using the option 'full cheap erase' to avoid major problems. Is there another way to format the flight controller?

Thank you!

**Comment by teknoflyer** [comment]
I tried my spr f3 deluxe board and the cal worked without a problem. you must start with the board upright. when I tried any other way the program would freeze or it would just say the board was finished calibration without going through any steps. I also started it upright and then did the rest in various orders including upside down last. they all worked fine. the only real problem I had with this BG board was that I had to use the "Recovery / Lost communication" procedure every time in order to get it to flash. 
are you getting an ok on the mag icon and the preflight hardware health? the acc values show on the cal page after cal is complete but the compass values only show after a save & reboot. hope this helps.
